src/main/scala/cromwell/binding/types/WdlArrayType.scala

```diff
@@ -20,6 +21,15 @@ case class WdlArrayType(memberType: WdlType) extends WdlType {
       WdlArray(WdlArrayType(WdlFileType), wdlArray.value.map(str => WdlFile(str.asInstanceOf[WdlString].value)).toList)
     case wdlArray: WdlArray if wdlArray.wdlType.memberType == memberType => wdlArray
     case wdlArray: WdlArray if wdlArray.wdlType.memberType == WdlAnyType => coerceIterable(wdlArray.value)
+    case wdlArray: WdlArray if wdlArray.wdlType.memberType.isInstanceOf[WdlArrayType] && memberType.isInstanceOf[WdlArrayType] =>
+      val coercedElements = wdlArray.value.map(v => memberType.coerceRawValue(v))
+      val coercedArray = coercedElements.collect({case x: Failure[_] => x}) match {
+        case failures if failures.nonEmpty =>
```

Comment:
I can't remember offhand but there's probably something in scalaz (which we use now) to handle this via an applicative, which is effectively what your'e doing here.